src/include/units/quantity.h

```diff
@@ -324,7 +324,7 @@ template<typename D, typename U1, typename Rep1, typename U2, typename Rep2>
 
 template<typename D, typename U, typename Rep, Scalar Value>
 [[nodiscard]] constexpr Quantity AUTO operator*(const quantity<D, U, Rep>& q, const Value& v)
-  requires std::regular_invocable<std::multiplies<>, Rep, Value>
+  requires std::regular_invocable<std::multiplies<>, Rep, Value> && std::is_arithmetic_v<Value>
```

Comment:
`is_arithmetic` cannot be used here because it limits values only to built-in arithmetic types and not to all types that behave as such. Besides, not arithmetic operations are needed in this case so this heavily overconstrains the function template.